src/v/raft/group_manager.cc

```diff
@@ -79,6 +79,16 @@ ss::future<> group_manager::remove(ss::lw_shared_ptr<raft::consensus> c) {
       });
 }
 
+ss::future<> group_manager::shutdown(ss::lw_shared_ptr<raft::consensus> c) {
+    return c->stop()
+      .then(
+        [this, id = c->group()] { return _heartbeats.deregister_group(id); })
+      .finally([this, c] {
+          _groups.erase(
```

Comment:
Does this work? 

Doesn’t the std::remove *invalidate* the _groups.end() in case of relocation? 

I don’t think these integrators are stable. This looks like a bug to me.

Reply:
This is C++ idiom for erasing element from a vector, std::remove moves all elements that aren't going to be removed to the beginning of a range and returns an iterator to first element that is going to be removed, `std::erase` do the actual erasure. https://en.cppreference.com/w/cpp/algorithm/remove. We can replcate it with `std::erase` it is available since C++20

Reply:
yeh this looks ok